hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/DiscoveryJoiner.java

```diff
@@ -54,7 +54,7 @@ public DiscoveryJoiner(Node node, DiscoveryService discoveryService, boolean use
     @Override
     protected Collection<Address> getPossibleAddressesForInitialJoin() {
         long deadLine = System.nanoTime() + SECONDS.toNanos(maximumWaitingTimeBeforeJoinSeconds);
-        for (int i = 0; System.nanoTime() < deadLine; i++) {
+        for (int i = 0; System.nanoTime() - deadLine < 0; i++) {
```

Comment:
I'm not sure about this one. Please pull this out of this PR because it isn't related to timing and lets run some ad hoc examples to make sure we don't run into some nasty overflow/sign related problems.